weblate/trans/models/unit.py

```diff
@@ -1868,11 +1867,10 @@ def translate(
         self.update_translation_memory(user)
 
         if change_action == ActionEvents.AUTO:
-            self.labels.add(component.project.automatically_translated_label)
+            self.automatically_translated = True
         else:
-            self.labels.through.objects.filter(
-                unit=self, label__name="Automatically translated"
-            ).delete()
+            self.automatically_translated = False
+        self.save(update_fields=["automatically_translated"])
```

Comment:
Can't we move this above just before the `save_backend` call to avoid additional database hit?